src/core/ekat_assert.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef EKAT_ASSERT_HPP
```

Comment:
This file is the same as before, but the KERNEL macros have been moved in the KokkosUtils target, and I purged the CONSTEXPR_ASSERT (which was meant to circumvent C++11 limitations, but was never used anyways)